Eastern-Research-Group/expert-query
# Feature/254 fix cypress tests

## Related Issues:
* https://jira.epa.gov/browse/EQ-254

## Main Changes:
* Fixed the cypress tests. Most of them were broken due to the api route changes and some were broken due to UI changes.
* Made the tests run faster, by updating the `selectOption` custom command. Now the tests run in about 8 minutes instead of 24 minutes. 
  * Before we had a delay of 0.5 seconds between typing each character in the dropdown, this was the primary reason for the slowness.
* The `copy-box.cy.ts` tests were failing intermittently. I beleive this was due to updating cypress recently. I ended up having to pull in another depency and using `realClick` to fix this. Here is where I found the solution https://github.com/cypress-io/cypress/issues/18198 
* I updated the tests using the cloud.gov database. This should reduce the test maintenance, since the full dataset will always be available on the dev database. It seemed like everytime the oracle => s3 part of the etl ran our tests would break, because the data returned from the ords services would change. 

## Steps To Test:
1. Run `npm ci` from the app folder
2. Point the app to the cloud.gov dev database
3. Run `npm run coverage` verify that all tests pass

## TODO:
- [ ] Add tests for new features, such as the national downloads and api documentation pages. We can do this if we have some left over hours, otherwise we should wait until version 2.
